nifi-toolkit/nifi-toolkit-encrypt-config/src/test/groovy/org/apache/nifi/properties/ConfigEncryptionToolTest.groovy

```diff
@@ -0,0 +1,1225 @@
+        File workingFile = new File("tmp_nifi.properties")
+        workingFile.delete()
+
+        final List<String> originalLines = inputPropertiesFile.readLines()
+
+        ConfigEncryptionTool tool = new ConfigEncryptionTool()
+        String[] args = ["-n", inputPropertiesFile.path, "-o", workingFile.path, "-k", KEY_HEX]
+        tool.parse(args)
+        NiFiProperties niFiProperties = tool.loadNiFiProperties()
+        tool.@niFiProperties = niFiProperties
+        logger.info("Loaded ${niFiProperties.size()} properties from ${inputPropertiesFile.path}")
+
+        // Act
+        tool.writeNiFiProperties()
+        logger.info("Wrote to ${workingFile.path}")
+
+        // Assert
+        final List<String> updatedLines = workingFile.readLines()
+        niFiProperties.getPropertyKeys().every { String key ->
+            assert updatedLines.contains("${key}=${niFiProperties.getProperty(key)}".toString())
+        }
+
+        assert originalLines == updatedLines
+
+        logger.info("Updated nifi.properties:")
+        logger.info("\n" * 2 + updatedLines.join("\n"))
+
+        workingFile.deleteOnExit()
+    }
+
+    @Test
+    void testWriteNiFiPropertiesShouldHandleWriteFailureWhenFileExists() {
+        // Arrange
+        File inputPropertiesFile = new File("src/test/resources/nifi_with_sensitive_properties_unprotected.properties")
+        File workingFile = new File("tmp_nifi.properties")
+        workingFile.delete()
+
+        Files.copy(inputPropertiesFile.toPath(), workingFile.toPath())
+        // Read-only set of permissions
+        Files.setPosixFilePermissions(workingFile.toPath(), [PosixFilePermission.OWNER_READ, PosixFilePermission.GROUP_READ, PosixFilePermission.OTHERS_READ] as Set)
+        logger.info("Set POSIX permissions to ${Files.getPosixFilePermissions(workingFile.toPath())}")
+
+        ConfigEncryptionTool tool = new ConfigEncryptionTool()
+        String[] args = ["-n", inputPropertiesFile.path, "-o", workingFile.path, "-k", KEY_HEX]
+        tool.parse(args)
+        NiFiProperties niFiProperties = tool.loadNiFiProperties()
+        tool.@niFiProperties = niFiProperties
+        logger.info("Loaded ${niFiProperties.size()} properties from ${inputPropertiesFile.path}")
+
+        // Act
+        def msg = shouldFail(IOException) {
+            tool.writeNiFiProperties()
+            logger.info("Wrote to ${workingFile.path}")
+        }
+        logger.expected(msg)
+
+        // Assert
+        assert msg == "The nifi.properties file at ${workingFile.path} must be writable by the user running this tool".toString()
+
+        workingFile.deleteOnExit()
+    }
+
+    @Test
+    void testWriteNiFiPropertiesShouldHandleWriteFailureWhenFileDoesNotExist() {
+        // Arrange
+        File inputPropertiesFile = new File("src/test/resources/nifi_with_sensitive_properties_unprotected.properties")
+        File tmpDir = new File("tmp/")
```

Comment:
I think we should make all these references to "tmp/" be "target/tmp/", right now it leaves a "tmp" directory at the root of nifi-toolkit-encrypt-config after running a build